src/Microsoft.AspNet.Authentication.OpenIdConnect/Events/RedirectFromIdentityProviderContext.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Http;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+
+namespace Microsoft.AspNet.Authentication.OpenIdConnect
+{
+    public class RedirectFromIdentityProviderContext : BaseControlContext<OpenIdConnectAuthenticationOptions>
```

Comment:
Hum, I'm not sure it's a really pertinent name then, since the ASP.NET 5 is always the initiating party when calling `Context.Authentication.SignOutAsync`: you're indeed redirected "from" the identity provider if you provided a `post_logout_redirect_uri` and if the identity provider agreed to redirect the user agent, but this doesn't happen at the same stage.